stdlib/public/Concurrency/Actor.cpp

```diff
@@ -1031,6 +1031,8 @@ static Job *
 preprocessQueue(JobRef unprocessedStart, JobRef unprocessedEnd, Job *existingProcessedJobsToMergeInto)
 {
   assert(existingProcessedJobsToMergeInto != NULL);
+  assert(unprocessedStart.needsPreprocessing());
+  assert(unprocessedStart.getAsJob() != unprocessedEnd.getAsJob());
```

Comment:
We tripped on assert on line 1054 in CI. The only way we could have NULL in newProcessedJobs was if we had no jobs to process which means `unprocessedStart == unprocessedEnd`.

And I thought shouldn't happen because we come into this code path when the compare_exchange_weak fails. Except compare_exchange_weak can fail spuriously and so we ended up tripping on an overly strong assert. I added a short-circuit in `drainOne` to detect this spurious failure and handle it and only go through the preprocess logic again if we actually did run into a concurrent enqueuer.

Also, more asserts.